kernel/mmu.c

```diff
@@ -348,10 +348,9 @@ static int map_anon_page(void *addr, uint32_t flags)
 void *k_mem_map(size_t size, uint32_t flags)
```

Comment:
Does it make sense returning the pointer to the usable memory (`dst + CONFIG_MMU_PAGE_SIZE`) directly instead of relying on the user to add the offset anyway?

Reply:
On the modified code, line 511, it does add the offset to `dst`. The new `k_mem_map()` introduces two guard pages (which are unmapped). The function is supposed to return a usable pointer. Or else, developer might use the return value directly and hit the guard page.

Reply:
Ohhh, right, for some reason I didn't see it. LGTM.